check/test.go

```diff
@@ -342,17 +342,36 @@ func (ts *tests) execute(s string) *testOutput {
 	return finalOutput
 }
 
-func toNumeric(a, b string) (c, d int) {
+type exiter interface {
+	Exit(code int)
+}
+
+type osExit struct{}
+
+var oe *osExit
+
+func newOsExit() *osExit {
+	if oe == nil {
+		oe = &osExit{}
+	}
+	return oe
+}
+
+func (eh *osExit) Exit(code int) {
+	os.Exit(code)
+}
+
+func toNumeric(a, b string, e exiter) (c, d int) {
```

Comment:
I'm thinking whether the exiter abstraction is absolutely necessary at this level. 

I guess that testing `toNumeric` method was challenging due to `s.Exit` call and `fmt.Fprintf` to the std out. Both are modifying so called global state which, in general, we're trying to avoid in unit tests.

What if we've changed the signature to return an error as 3rd param and let the caller of `toNumeric` decide whether to `os.Exit` or not?

```go
func toNumeric(a, b string) (c, d int, err error) {
	c, err = strconv.Atoi(strings.TrimSpace(a))
	if err != nil {
		return 0, 0, fmt.Errorf("converting %s to numeric: %v", a, err)
	}
	d, err = strconv.Atoi(strings.TrimSpace(b))
	if err != nil {
		return 0, 0, fmt.Errorf("converting %s to numeric: %v", b, err)
	}

	return c, d, nil
}
```

By doing that you could write somehow lighter tests:

```go
_, _, err := toNumeric("NaN", 3)
assert.EqualError(t, err, "converting NaN to numeric: some underlying error")
_, _, err := toNumeric(5, "NaN")
assert.EqualError(t, err, "converting NaN to numeric: some underlying error")
```

I believe it's more idiomatic and gives you an opportunity to assert on error messages, which is very important for debugging and troubleshooting.

Reply:
I like your approach better than mine. 💯